docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -1,15 +1,158 @@
 ---
 title: Create an event
 description:
-  This guide explains how to create an event in Front-Commerce and dispatch it to external services.
+  In this guide, we will cover the creation of an emitter and how to implement
+  them in your pages, we will create an Emitter that Emit an event when a
+  customer reach a page called FAQ.
 sidebar_position: 1
 ---
 
 <p>{frontMatter.description}</p>
 
-:::info WIP
+## What is an event emitter
 
-- Create an event emitter
-- Emit an event from your code
+The events are dispatched through the use of an event emitter. The role of the
+event emitter is to push events to the server event pipeline to be processed by
+the server.
+
+Emitter are implemented in Domain Event class that define an event related to a
+specific event for your app.
+
+## Prerequisites
+
+Before diving into the code, ensure you have the following prerequisites in
+place:
+
+- A Front-Commerce project configured and ready to use
+- [FAQ GraphQL Module](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/tree/main/skeleton/example-extensions/faq-demo/graphql)
+  installed and configured if you want to execute code samples without any
+  changes
+
+## Create a Domain event class
+
+Let's create a file in our project, called `QuestionVotedEvent.ts`:
+
+```tsx title="app/server-events/events/QuestionVotedEvent.ts"
+import { ServerEvent } from "@front-commerce/core/events";
+
+// We define a static event Name for our event
+const QuestionVotedEventName = "QuestionVotedEvent";
+
+// Definition of the payload
+type QuestionVotedEventPayload = {
+  questionSlug: string;
+  upVote: boolean;
+};
+
+// We define an event class that extends the ServerEvent class
+// This class will handle data to be pushed to the server event pipeline
+export default class QuestionVotedEvent extends ServerEvent<QuestionVotedEventPayload> {
+  static event_type = QuestionVotedEventName;
+
+  // We define the constructor for our event
+  // It will call the super constructor (constructor of ServerEvent)
+  // to dispatch event to the server event pipeline
+  constructor(questionSlug: string, upVote: boolean, shopId: string) {
+    super(
+      QuestionVotedEventName,
+      new Date(),
+      // This are data that we want to gather
+      // The question
+      // If the question has been upvoted or downvoted
+      { questionSlug, upVote },
+      // This are the event metadata, they are relative to the event
+      // but don't contain event data, here we collect the shopId
+      // to determine on which shop the question has been voted
+      { shopId }
+    );
+  }
+}
+```
+
+:::info
+
+You can checkout the
+[ServerEvent Reference](/docs/3.x/api-reference/front-commerce-core/ServerEvent)
+to learn more about the usage of ServerEvent class.
 
 :::
+
+That's it!
+
+## Use the Event
+
+Now that we've created a Domain event class, let's use it in a route action:
+
+```tsx title="app/routes/_main.faq.$slug.tsx"
```

Comment:
suggestion: Please see what I suggested in https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/merge_requests/3163 about this file.